src/lxc/cgroups/cgfsng.c

```diff
@@ -554,12 +598,9 @@ static bool cpuset1_cpus_initialize(int dfd_parent, int dfd_child,
 {
 	__do_free char *cpulist = NULL, *fpath = NULL, *isolcpus = NULL,
 		       *offlinecpus = NULL, *posscpus = NULL;
-	__do_free uint32_t *isolmask = NULL, *offlinemask = NULL,
-			   *possmask = NULL;
+	__do_free uint32_t *possmask = NULL;
 	int ret;
-	size_t isol_last_set_bit = 0, offline_last_set_bit = 0,
-	       poss_last_set_bit = 0;
-	bool flipped_bit = false;
+	size_t poss_last_set_bit = 0;
 
 	posscpus = read_file_at(dfd_parent, "cpuset.cpus", PROTECT_OPEN, 0);
```

Comment:
I don't think it is a foregone conclusion that cpuset.cpus file provides all possible cpus.  In a system I am looking at comprised of 448 cpus viewing /sys/fs/cgroup/cpuset/cpuset.cpus it lists only the on-line cpus which happens to be 0-103.  /sys/devices/system/cpu/possible actually provides the entire range of cpus.

Reply:
But that shouldn't be relevant. We only care about the cpus that are available according to `cpuset.cpus`. Where would "possible" be relevant?

Reply:
Ahh.  I did not notice that lxc_cpumask() was changed and added the third argument for last_set_bit so in the case I cited above it would be a NOP as there are no offlines within the range of the onlines.  And the new code handles cases where offlines would be within the range of cpuset.cpus.